airbyte-integrations/bases/connector-acceptance-test/unit_tests/test_asserts.py

```diff
@@ -119,3 +119,82 @@ def test_validate_records_format(record, configured_catalog, valid):
+            {"type": "object", "additionalProperties": False, "properties": {"a": {"type": "string"}}},
+        ),
+        (
+            {"type": "object", "properties": {"a": {"type": "string"}, "some_obj": {"type": ["null", "object"]}}},
+            {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": {"a": {"type": "string"}, "some_obj": {"type": ["null", "object"]}},
+            },
+        ),
+        (
+            {
+                "type": "object",
+                "properties": {"a": {"type": "string"}, "some_obj": {"type": ["null", "object"], "properties": {"a": {"type": "string"}}}},
+            },
+            {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": {
+                    "a": {"type": "string"},
+                    "some_obj": {"additionalProperties": False, "properties": {"a": {"type": "string"}}, "type": ["null", "object"]},
+                },
+            },
+        ),
+        (
+            {"type": "object", "properties": {"a": {"type": "string"}, "b": {"type": "array", "items": {"type": "string"}}}},
+            {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": {"a": {"type": "string"}, "b": {"type": "array", "items": {"type": "string"}}},
+            },
+        ),
+        (
+            {"type": "object", "properties": {"a": {"type": "string"}, "b": {"type": "array", "items": {"type": "object"}}}},
+            {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": {"a": {"type": "string"}, "b": {"type": "array", "items": {"type": "object"}}},
+            },
+        ),
+        (
+            {
+                "type": "object",
+                "properties": {
+                    "a": {"type": "string"},
+                    "b": {"type": "array", "items": {"type": "object", "properties": {"a": {"type": "string"}}}},
+                },
+            },
+            {
+                "type": "object",
+                "additionalProperties": False,
+                "properties": {
+                    "a": {"type": "string"},
+                    "b": {
+                        "type": "array",
+                        "items": {"type": "object", "additionalProperties": False, "properties": {"a": {"type": "string"}}},
+                    },
+                },
+            },
+        ),
+    ],
+    ids=[
+        "simple_schema",
+        "schema_with_object_without_properties",
+        "schema_with_object_with_properties",
+        "schema_with_array_of_strings",
+        "schema_with_array_of_objects",
+        "schema_with_array_of_objects_with_properties",
+    ],
+)
+def test_validate_additional_properties(original_schema, expected_schema):
+    assert _enforce_false_additional_properties(original_schema) == expected_schema
```

Comment:
It is weird to me to test a private methods. There is nothing enforcing the public method `verify_records_schema` to use `_enforce_false_additional_properties` hence, the behavior of the public interface might be very different than what we expect given this suite of tests; we have no guarantee that we are testing that there are no additional properties.

Can we have the same tests setup but call `verify_records_schema` instead and test the public behavior?

Reply:
test updated, please take a look